lib/plugins/aws/package/compile/events/alb/index.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+const validate = require('./lib/validate');
+const compileTargetGroups = require('./lib/targetGroups');
+const compileListenerRules = require('./lib/listenerRules');
+const compilePermissions = require('./lib/permissions');
+
+class AwsCompileAlbEvents {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+
+    Object.assign(
+      this,
+      validate,
+      compileTargetGroups,
+      compileListenerRules,
+      compilePermissions
+    );
+
+    this.hooks = {
+      'package:compileEvents': () => {
+        this.validated = this.validate();
+
+        return BbPromise.try(() => {
```

Comment:
I think this should wrap whole body (so also include `this.validated = this.validate();` line)